### ISSUE TITLE
New LazyFrames is broken

### ISSUE THREAD

**Issue by GarrettAndersen** [openai/baselines#295]
The updated LazyFrames function uses way more memory than before.

**Comment by benbotto** [comment]
Agreed.  Looks like it's going to take nearly 30 GB to store the 1 million frames needed to reproduce the Nature paper on DQN (that's a rough figure).  It should only need 2^20 * 84^2 / 1024^3 = 6.89 GB when using PER.  I'm pretty sure it's because of the way `np.concatenate` is being used in `LazyFrames`  (https://github.com/openai/baselines/blob/master/baselines/common/atari_wrappers.py#L197).  That's going to duplicate the frames over and over.  If instead `FrameStack` used a shape of `(4, 84, 84)`, then simply returning `[frame for frame in self.frames]`--and not using `LazeFrames`--would fix the memory problem.  The convolutional layers would then need to have a channels-first format, though (`data_format="NCHW"`).
